internal/scrapertest/compare_test.go

```diff
@@ -334,6 +334,20 @@ func TestCompareMetrics(t *testing.T) {
 				reason: "An unpredictable data point value will cause failures if not ignored.",
 			},
 		},
+		{
+			name: "ignore-single-metric",
+			compareOptions: []CompareOption{
+				IgnoreMetricValues("sum.two"),
+			},
+			withoutOptions: expectation{
+				err: multierr.Combine(
+					errors.New("datapoints for metric: `sum.two`, do not match expected"),
+					errors.New("datapoint with attributes: map[], does not match expected"),
```

Comment:
Just curios, why is this error raised?

Reply:
you mean specifically the `attribute: map[]` error? I'm not sure, i copied from the previous test

Reply:
Yes, it's seems weird given that the are no attributes in the testdata

Reply:
The message could be more clear, but basically it's just identifying which data point was found to not match the expected. In this case, there's only one data point and it has no attributes (hence `with attributes: map[]`) but in a scenario with multiple data points the map would be rendered with key/value pairs.

Reply:
Sorry, I still don't understand :) Looks like both _expected_ and _got_ datapoints have empty attributes in this tests case, but the error is raised.

Reply:
It's not the attributes that are wrong. Something is wrong with another aspect of the data point that has this (empty) set of attributes. This error is supposed to tell you _which_ data point is wrong, not _what_ is wrong with it.

Reply:
Anyway. nvm. I understand now. It makes sense to keep message per metric and par datapoint 👍 . Thanks for explaining